### PR TITLE
Fix for bug in Show Detectors when Efixed not available

### PR DESCRIPTION
**Pull request by SilkeSchomann**
#### Summary of work

In some cases when the `efixed` value is not available for a detector, an `invalid_argument` exception is thrown instead of a `runtime_error`. Previously only the `runtime_error` exception was caught and the creation of the detector table failed. Now both types of exception are caught and `Show Detectors` works regardless of whether the `efixed` value is available or not.

Fixes #[36988](https://github.com/mantidproject/mantid/issues/36988)

**Report to:** @mducle 

### To test:

Follow the instructions in the original issue: https://github.com/mantidproject/mantid/issues/36988

In addition, load a few old and new data sets from different instruments and check whether `Show Detectors` works as expected.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by adriazalvarez — CHANGES_REQUESTED** [review]
Tested with LET data. It is possible to create a detector table with no crashes. 
I would suggest that perhaps it should be good to add a unit test against this exception (invalid argument)?

**Review by adriazalvarez — DISMISSED** [review]
Thanks for the changes!

File: Framework/Algorithms/test/CreateDetectorTableTest.h [hunk 1]
```diff
@@ -98,6 +100,11 @@ class CreateDetectorTableTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS_NOTHING(alg.execute();)
     TS_ASSERT(alg.isExecuted());
 
+    // Check that a missing efixed value throws
+    const auto &spectrumInfo = inputWS->spectrumInfo();
+    std::shared_ptr<const IDetector> detector(&spectrumInfo.detector(0), Mantid::NoDeleting());
+    TS_ASSERT_THROWS_ANYTHING(inputWS->getEFixed(detector));
```
Comment (thomashampson) [inline Framework/Algorithms/test/CreateDetectorTableTest.h:106]: I don't think this is testing your bug fix, it would still pass without the code changes.

**Review by thomashampson — CHANGES_REQUESTED** [review]

**Review by adriazalvarez — APPROVED** [review]
Thanks for fixing the test. I think it should be good now.

**Review by thomashampson — APPROVED** [review]
Fixes a bug reported by user.

**Pull request merged**